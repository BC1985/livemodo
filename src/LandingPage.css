h1 {
  padding-top: 100px;
}

.description {
  margin-left: 50px;
  margin-right: 50px;
  margin-top: 20px;
  margin-bottom: 100px;
  line-height: 25px;
}

.call-to-action {
  padding: 20px;
}
