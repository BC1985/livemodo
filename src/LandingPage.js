import React, { Component } from "react";
// import { Router, Link } from "react-router-dom";
// import LoginPage from "./LoginPage";
const Description = () => {
  return (
    <div>
      Have you ever bought a ticket to see your favorite band play live in your
      hometown only to have your experience ruined by the lead singer's lack of
      stage presence? Or maybe you're kicking yourself for missing out on
      catching that unknown band play at the block party right before they made
      it big and now the concert tickets are too expensive... If so, then
      Livemodo is the right tool for any music lover. Houndreds of reviews from
      fellow concertgoers will help you manage your expectations in your
      upcoming concert escapades.
    </div>
  );
};

export default class LandingPage extends Component {
  constructor() {
    super();
    this.state = {
      isLoggedIn: true,
      userName: "John"
    };
  }

  UserIsLoggedIn = () => {
    return <p>Welcome {this.state.userName}!</p>;
  };
  UserNotLoggedIn = () => {
    return (
      // <Router>
      //   <Link to="/login">
      <p>Log in</p>
      //   </Link>
      // </Router>
    );
  };

  render() {
    const Greeting = () => {
      if (this.state.isLoggedIn) {
        return <this.UserIsLoggedIn />;
      }
      return <this.UserNotLoggedIn />;
    };

    return (
      <div className="container">
        <div className="header">
          <h1>Livemodo</h1>
          <h3>Your wiki live shows resource</h3>
        </div>
        <div className={!this.state.isLoggedIn ? "login" : "greeting"}>
          <Greeting />
        </div>
        <div className="description">
          <Description />
        </div>
      </div>
    );
  }
}
