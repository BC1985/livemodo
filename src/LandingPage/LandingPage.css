/* Global settings */

#greeting {
  text-align: center;
  font-family: "Josefin Sans", sans-serif;
}

.description {
  margin: 0 20% 10% 20%;
  line-height: 25px;
  text-align: left;
}

.header {
  color: whitesmoke;
  text-align: center;
  font-family: "Josefin Sans", sans-serif;
}
.header > h1 {
  text-shadow: 4px 1px 0 rgba(0, 0, 0, 0.3);
}
#greeting {
  color: whitesmoke;
}

/* ----------- iPhone 5, 5S, 5C and 5SE ----------- */

/* Portrait */
@media only screen and (min-device-width: 320px) and (max-device-width: 568px) and (-webkit-min-device-pixel-ratio: 2) and (orientation: portrait) {
  .container {
    display: flex;
    flex-direction: column;
    background-size: cover;
    background-repeat: no-repeat;
    min-width: 100%;
  }

  .description {
    margin: 0 20%;
    line-height: 25px;
  }
  .header {
    margin: 5% 20% 0 20%;
  }
  .landing-page-push {
    height: 60px;
  }
}
/* Landscape */
@media only screen and (min-device-width: 320px) and (max-device-width: 568px) and (-webkit-min-device-pixel-ratio: 2) and (orientation: landscape) {
  .header {
    margin-top: 50px;
  }
}

/* ----------- iPhone 6, 6S, 7 and 8 ----------- */

/* Landscape */
@media only screen and (min-device-width: 375px) and (max-device-width: 667px) and (-webkit-min-device-pixel-ratio: 2) and (orientation: landscape) {
  .container {
    background-image: url(../images/Resized/concert.png);
    background-repeat: no-repeat;
    background-size: cover;
    display: flex;
    flex-direction: column;
  }
  .header {
    margin-top: 50px;
  }

  .description {
    margin: 0 20%;
    line-height: 25px;
  }
  .landing-page-push {
    height: 60px;
  }
}

/* Portrait */
@media only screen and (min-device-width: 375px) and (max-device-width: 667px) and (-webkit-min-device-pixel-ratio: 2) and (orientation: portrait) {
  h1 {
    padding-top: 30px;
  }

  .container {
    text-align: center;
  }
  .header {
    color: whitesmoke;
  }

  #login {
    padding: 20px;
    font-family: "Josefin Sans", sans-serif;
    text-decoration: underline;
    color: whitesmoke;
  }

  #login-button > button {
    padding: 20px;
    border-radius: 5px;
    width: 100px;
    border: 3px solid gray;
  }
}

/* ----------- iPhone 6+, 7+ and 8+ ----------- */

/* Landscape */
@media only screen and (min-device-width: 414px) and (max-device-width: 736px) and (-webkit-min-device-pixel-ratio: 3) and (orientation: landscape) {
  .container {
    display: flex;
    flex-direction: column;
    background-attachment: local;
  }
  .header {
    margin-top: 10%;
  }
}

/* ----------- iPhone X --------------- */

/* Portrait */
@media only screen and (min-device-width: 375px) and (max-device-width: 812px) and (-webkit-min-device-pixel-ratio: 3) and (orientation: portrait) {
  #button {
    margin-bottom: 20px;
  }
  #greeting {
    padding-bottom: 0px;
  }
  .landing-page-push {
    height: 100px;
  }
}
/* Landscape */
@media only screen and (min-device-width: 375px) and (max-device-width: 812px) and (-webkit-min-device-pixel-ratio: 3) and (orientation: landscape) {
  .container {
    background-image: url(../images/Resized/concert.png);
    background-repeat: no-repeat;
    background-size: cover;

    display: flex;
    flex-direction: column;
  }
  .header {
    margin-top: 5%;
  }
}
/* Samsung Galaxy S6, S6+, S7, S7+ | 1440Ã—2560 pixels ----------- */

/* Portrait */
@media screen and (device-width: 360px) and (device-height: 640px) and (-webkit-device-pixel-ratio: 4) and (orientation: portrait) {
  .header {
    margin-top: 30%;
  }
  .container {
    background-image: url(../images/Resized/concert.png);
    background-repeat: no-repeat;
    background-size: cover;

    display: flex;
    flex-direction: column;
  }
}

/* ----------- iPad 3, 4 and Pro 9.7" ----------- */

/* Portrait */
@media only screen and (min-device-width: 768px) and (max-device-width: 1024px) and (orientation: portrait) and (-webkit-min-device-pixel-ratio: 2) {
  .container {
    display: block;
    width: 100%;
    background: url(../images/Resized/stage-small.png);
    background-size: cover;
    background-repeat: no-repeat;
    overflow: hidden;

    animation: slide 20s infinite;
  }

  @keyframes slide {
    25% {
      background-image: url(../images/Resized/singer.png);
      background-position: center;
    }
    50% {
      background-image: url(../images/Resized/indie-band.png);
      background-position: top;
    }
    75% {
      background-image: url(../images/Original-size/audience-backlit-blue-2078071.jpg);
      background-position: left;
    }
    100% {
      background-image: url(../images/Resized/concert.png);
      background-position: center;
    }
  }

  .header {
    padding-top: 40%;
  }
  .description,
  .header {
    line-height: 35px;
  }

  .landing-page-push {
    height: 400px;
  }
}

/* Landscape */
@media only screen and (min-device-width: 768px) and (max-device-width: 1024px) and (orientation: landscape) and (-webkit-min-device-pixel-ratio: 2) {
  .container {
    background-image: url(../images/Resized/concert.png);
    background-size: cover;
    background-position: center;
  }
}
/* ----------- iPad Pro 10.5" ----------- */

/* Portrait and Landscape */
@media only screen and (min-device-width: 834px) and (max-device-width: 1112px) and (-webkit-min-device-pixel-ratio: 2) {
}

/* Portrait */

@media only screen and (min-device-width: 834px) and (max-device-width: 834px) and (orientation: portrait) and (-webkit-min-device-pixel-ratio: 2) {
}

/* Landscape */

@media only screen and (min-device-width: 1112px) and (max-device-width: 1112px) and (orientation: landscape) and (-webkit-min-device-pixel-ratio: 2) {
  .landing-page-push {
    height: 50%;
  }
}

/* ----------- iPad Pro 12.9" ----------- */

/* Portrait */

@media only screen and (min-device-width: 1024px) and (max-device-width: 1366px) and (orientation: portrait) and (-webkit-min-device-pixel-ratio: 2) {
  .landing-page-push {
    height: 60%;
  }
}

/* Landscape */

@media only screen and (min-device-width: 1366px) and (max-device-width: 1366px) and (orientation: landscape) and (-webkit-min-device-pixel-ratio: 2) {
}

/* ----------Laptops and desktops----------- */

/* ------- Non-Retina Screens ------- */
@media screen and (min-device-width: 1024px) and (max-device-width: 1600px) and (-webkit-min-device-pixel-ratio: 2) {
  .container {
    display: block;
    /* font-size: 1.2em; */
    width: 100%;
    height: 100%;
    background: url(../images/Resized/stage-small-B&W.png);
    background-position: top;
    background-size: cover;
    overflow: hidden;
    -webkit-animation: slide 25s ease-in infinite;
    animation: slide 25s ease-in infinite;
  }
  .header {
    padding-top: 25%;
  }
  .description {
    font-size: 0.8em;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    justify-content: center;
    margin: 0 30% 10% 30%;
  }

  .landing-page-push {
    height: 350px;
  }

  @-webkit-keyframes slide {
    25% {
      background-image: url(../images/Resized/singer.png);
      background-position: center;
    }
    50% {
      background-image: url(../images/Resized/indie-band.png);
      background-position: top;
      background-size: cover;
    }
    75% {
      background-image: url(../images/Resized/large-crowd.png);
      background-size: cover;
      background-position: left;
    }
    100% {
      background-image: url(../images/Resized/concert.png);
      background-position: bottom;
    }
  }

  @keyframes slide {
    25% {
      background-image: url(../images/Resized/singer.png);
      background-position: center;
    }
    50% {
      background-image: url(../images/Resized/indie-band.png);
      background-position: top;
      background-size: cover;
    }
    75% {
      background-image: url(../images/Resized/large-crowd.png);
      background-size: cover;
      background-position: left;
    }
    100% {
      background-image: url(../images/Resized/concert.png);
      background-position: bottom;
    }
  }
}

/* ----------- Retina Screens ----------- */
@media screen and (min-device-width: 1200px) and (max-device-width: 1600px) and (-webkit-min-device-pixel-ratio: 2) and (min-resolution: 192dpi) {
}
@media screen and (min-device-width: 1600px) {
  .container {
    display: block;
    width: 100%;
    height: 100%;
    background: url(../images/Resized/stage-small-B&W.png);
    background-position: top;
    background-size: cover;
    overflow: hidden;
    -webkit-animation: slide 15s ease-in infinite;
    animation: slide 15s ease-in infinite;
  }
  .header {
    padding-top: 10%;
  }
  .description {
    line-height: 2em;
    padding: 20px;
    -webkit-box-pack: center;
    -ms-flex-pack: center;
    justify-content: center;
    margin: 0 30% 10% 30%;
  }

  .landing-page-push {
    height: 200px;
  }

  @-webkit-keyframes slide {
    25% {
      background-image: url(../images/Resized/singer.png);
      background-position: center;
    }
    50% {
      background-image: url(../images/Resized/indie-band.png);
      background-position: top;
      background-size: cover;
    }
    75% {
      background-image: url(../images/Resized/large-crowd.png);
      background-size: cover;
      background-position: left;
    }
    100% {
      background-image: url(../images/Resized/concert.png);
      background-position: bottom;
    }
  }

  @keyframes slide {
    25% {
      background-image: url(../images/Resized/singer.png);
      background-position: center;
    }
    50% {
      background-image: url(../images/Resized/indie-band.png);
      background-position: top;
      background-size: cover;
    }
    75% {
      background-image: url(../images/Resized/large-crowd.png);
      background-size: cover;
      background-position: left;
    }
    100% {
      background-image: url(../images/Resized/concert.png);
      background-position: bottom;
    }
  }
}
