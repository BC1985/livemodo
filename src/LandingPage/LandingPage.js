import React, { Component } from "react";
import { Redirect, Link } from "react-router-dom";
import Button from "../Button/Button";
import "./LandingPage.css";
import { TokenService } from "../utils/token-service";
const landingPageStyle = {
  color: "whiteSmoke"
};
const Description = () => {
  return (
    <div>
      <p style={landingPageStyle} role="contentinfo">
        Have you ever bought a ticket to see your favorite band play live in
        your hometown only to have your experience ruined by the lead singer's
        lack of stage presence? Or maybe you're kicking yourself for missing out
        on catching that unknown band play at the block party right before they
        made it big and now the concert tickets are too expensive... If so, then
        Livemodo is the right tool for any music lover. Houndreds of reviews
        from fellow concertgoers will help you manage your expectations in your
        upcoming concert escapades.
        <br />
        <span
          style={{
            fontWeight: "bold",
            color: "lightGray",
            textDecoration: "underline",
            fontStyle: "italic"
          }}
        >
          To log in with dummy credentials, enter username JohnSmith and
          password JohnSmith1!
        </span>
      </p>
    </div>
  );
};

export default class LandingPage extends Component {
  constructor() {
    super();
    this.state = {
      userName: "John"
    };
  }
  redirect = () => {
    this.setState({
      Redirect: true
    });
  };
  UserNotLoggedIn = () => {
    return (
      <Link to="/login">
        <Button id="login-button" onClick={this.redirect}>
          Log in
        </Button>
      </Link>
    );
  };
  render() {
    if (this.state.Redirect) {
      return <Redirect push to="/login" />;
    }
    return (
      <div className="container">
        <div className="header" role="banner">
          <h1>Livemodo</h1>
          <h2>Your wiki live shows resource</h2>
        </div>
        <div id="greeting">
          {TokenService.hasAuthToken() ? (
            <div id="welcome">Welcome!</div>
          ) : null}
        </div>

        <div className="description">
          <Description />
        </div>
        <div className="landing-page-push" />
      </div>
    );
  }
}
