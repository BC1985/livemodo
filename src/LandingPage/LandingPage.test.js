import React from "react";
import ReactDOM from "react-dom";
import LandingPage from "../LandingPage/LandingPage";
import renderer from "react-test-renderer";

it("renders without crashing", () => {
  const div = document.createElement("div");
  ReactDOM.render(<LandingPage />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it("renders UI as expected", () => {
  const tree = renderer.create(<LandingPage />).toJSON();
  expect(tree).toMatchSnapshot();
});
