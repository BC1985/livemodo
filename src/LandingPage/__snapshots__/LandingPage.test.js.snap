// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders UI as expected 1`] = `
<div
  className="container"
>
  <div
    className="header"
    role="banner"
  >
    <h1>
      Livemodo
    </h1>
    <h2>
      Your wiki live shows resource
    </h2>
  </div>
  <div
    id="greeting"
  />
  <div
    className="description"
  >
    <div>
      <p
        style={
          Object {
            "color": "whiteSmoke",
          }
        }
      >
        Have you ever bought a ticket to see your favorite band play live in your hometown only to have your experience ruined by the lead singer's lack of stage presence? Or maybe you're kicking yourself for missing out on catching that unknown band play at the block party right before they made it big and now the concert tickets are too expensive... If so, then Livemodo is the right tool for any music lover. Houndreds of reviews from fellow concertgoers will help you manage your expectations in your upcoming concert escapades.
        <br />
        <span
          style={
            Object {
              "color": "lightGray",
              "fontStyle": "italic",
              "fontWeight": "bold",
              "textDecoration": "underline",
            }
          }
        >
          To log in with dummy credentials, enter username JohnSmith and password JohnSmith1!
        </span>
      </p>
    </div>
  </div>
  <div
    className="landing-page-push"
  />
</div>
`;
